[CI] NdJsonCrossFileScalarObjectConflictIT testDefaultSettingsFailsOnCrossFileScalarObjectConflict failing

**Build Scans:**
- [elasticsearch-intake #42689 / part3](https://gradle-enterprise.elastic.co/s/m2u5y4fx5tzho)
- [elasticsearch-pull-request #160371 / part-3](https://gradle-enterprise.elastic.co/s/i66pmlpr6gxdy)
- [elasticsearch-pull-request #160360 / part-3](https://gradle-enterprise.elastic.co/s/hhqjdym2bwxyi)
- [elasticsearch-pull-request #160366 / part-3](https://gradle-enterprise.elastic.co/s/7bcqjp5uqrkda)
- [elasticsearch-pull-request #160360 / part-3](https://gradle-enterprise.elastic.co/s/nkvu2cqzolwu2)
- [elasticsearch-pull-request #160371 / part-3](https://gradle-enterprise.elastic.co/s/h5foyo3y6mhzq)
- [elasticsearch-pull-request #160361 / part-3](https://gradle-enterprise.elastic.co/s/2iolvpaun4hyc)
- [elasticsearch-pull-request #160367 / part-3](https://gradle-enterprise.elastic.co/s/dularidvnob72)
- [elasticsearch-pull-request #160378 / releasetestcheckesqlengine_release-tests](https://gradle-enterprise.elastic.co/s/om7r3dpuvggkw)
- [elasticsearch-pull-request #160341 / part-3](https://gradle-enterprise.elastic.co/s/4a3ijayx57txs)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.NdJsonCrossFileScalarObjectConflictIT.testDefaultSettingsFailsOnCrossFileScalarObjectConflict" -Dtests.seed=25289218776CC07F -Dtests.locale=haw-Latn-US -Dtests.timezone=Asia/Novosibirsk -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.NdJsonCrossFileScalarObjectConflictIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testDefaultSettingsFailsOnCrossFileScalarObjectConflict'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: must fail on the cross-file shape conflict, got: java.lang.IllegalArgumentException: local filesystem access via file:// is disabled; set the [esql.datasource.local_allowed_paths] node setting to one or more allowed root paths to enable it
	at org.elasticsearch.xpack.esql.datasources.StorageProviderRegistry.createProviderTrackingConsumedKeys(StorageProviderRegistry.java:191)
	at org.elasticsearch.xpack.esql.datasources.StorageProviderRegistry.createProvider(StorageProviderRegistry.java:181)
	at org.elasticsearch.xpack.esql.datasources.ExternalSourceResolver.resolveProvider(ExternalSourceResolver.java:806)
	at org.elasticsearch.xpack.esql.datasources.ExternalSourceResolver.resolveMultiFileSource(ExternalSourceResolver.java:535)
	at org.elasticsearch.xpack.esql.datasources.ExternalSourceResolver.resolveSourceInner(ExternalSourceResolver.java:453)
	at org.elasticsearch.xpack.esql.datasources.ExternalSourceResolver.resolveSource(ExternalSourceResolver.java:434)
	a
[truncated]
```

**Issue Reasons:**
- [main] 66 consecutive failures in test testDefaultSettingsFailsOnCrossFileScalarObjectConflict
- [main] 2 consecutive failures in step part3
- [main] 43 consecutive failures in step part-3
- [main] 20 consecutive failures in step releasetestcheckesqlengine_release-tests
- [main] 1 consecutive failures in step checkpart3_release-tests
- [main] 2 consecutive failures in pipeline elasticsearch-intake
- [main] 26 consecutive failures in pipeline elasticsearch-pull-request
- [main] 66 failures in test testDefaultSettingsFailsOnCrossFileScalarObjectConflict (97.1% fail rate in 68 executions)
- [main] 2 failures in step part3 (100.0% fail rate in 2 executions)
- [main] 43 failures in step part-3 (97.7% fail rate in 44 executions)
- [main] 20 failures in step releasetestcheckesqlengine_release-tests (95.2% fail rate in 21 executions)
- [main] 2 failures in pipeline elasticsearch-intake (100.0% fail rate in 2 executions)
- [main] 26 failures in pipeline elasticsearch-pull-request (96.3% fail rate in 27 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -871,9 +871,3 @@ tests:
 - class: org.elasticsearch.test.rest.yaml.RcsCcsCommonYamlTestSuiteIT
   method: initializationError
   issue: https://github.com/elastic/elasticsearch/issues/152849
-- class: org.elasticsearch.xpack.esql.action.NdJsonCrossFileScalarObjectConflictIT
-  method: testDefaultSettingsFailsOnCrossFileScalarObjectConflict
-  issue: https://github.com/elastic/elasticsearch/issues/152865
-- class: org.elasticsearch.xpack.esql.action.NdJsonCrossFileScalarObjectConflictIT
-  method: testSkipRowPolicyNullFillsAndWarnsClient
-  issue: https://github.com/elastic/elasticsearch/issues/152866

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/NdJsonCrossFileScalarObjectConflictIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/NdJsonCrossFileScalarObjectConflictIT.java
@@ -17,7 +17,9 @@ import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.plugins.Plugin;
 import org.elasticsearch.transport.TransportService;
 import org.elasticsearch.xpack.core.esql.action.ColumnInfo;
+import org.elasticsearch.xpack.esql.datasource.http.HttpDataSourcePlugin;
 import org.elasticsearch.xpack.esql.datasource.ndjson.NdJsonDataSourcePlugin;
+import org.elasticsearch.xpack.esql.datasources.ExternalSourceSettings;
 import org.elasticsearch.xpack.esql.datasources.dataset.DeleteDatasetAction;
 import org.elasticsearch.xpack.esql.datasources.dataset.PutDatasetAction;
 import org.elasticsearch.xpack.esql.datasources.datasource.DeleteDataSourceAction;
@@ -100,14 +102,24 @@ public class NdJsonCrossFileScalarObjectConflictIT extends AbstractEsqlIntegTest
     @Override
     protected Collection<Class<? extends Plugin>> nodePlugins() {
         List<Class<? extends Plugin>> plugins = new ArrayList<>(super.nodePlugins());
+        plugins.add(HttpDataSourcePlugin.class);
         plugins.add(NdJsonDataSourcePlugin.class);
         plugins.add(TestDataSourcePlugin.class);
         return plugins;
     }
 
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
+        return Settings.builder()
+            .put(super.nodeSettings(nodeOrdinal, otherSettings))
+            .putList(ExternalSourceSettings.LOCAL_ALLOWED_PATHS.getKey(), createTempDir().getParent().toString())
+            .build();
+    }
+
     @Before
     public void requireFeatureFlag() {
         assumeTrue("requires external data sources feature flag", DatasetMetadata.ESQL_EXTERNAL_DATASOURCES_FEATURE_FLAG.isEnabled());
+        assumeTrue("requires local filesystem feature flag", HttpDataSourcePlugin.ESQL_EXTERNAL_DATASOURCES_LOCAL_FEATURE_FLAG.isEnabled());
     }
 
     /**

